File: server/src/main/java/com/fluffy/auth/domain/Member.java
package com.fluffy.auth.domain;

import com.fluffy.infra.persistence.AuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Member extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String email;

    @Column(nullable = false, columnDefinition = "VARCHAR(20)")
    @Enumerated(EnumType.STRING)
    private OAuth2Provider provider;

    @Column(nullable = false)
    private String socialId;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String avatarUrl;

    public Member(String email, OAuth2Provider provider, String socialId, String name, String avatarUrl) {
        this(null, email, provider, socialId, name, avatarUrl);
    }

    public Member(Long id, String email, OAuth2Provider provider, String socialId, String name, String avatarUrl) {
        this.id = id;
        this.email = email;
        this.provider = provider;
        this.socialId = socialId;
        this.name = name;
        this.avatarUrl = avatarUrl;
    }
}

File: server/src/main/kotlin/com/fluffy/auth/domain/Member.kt
package com.fluffy.auth.domain

import com.fluffy.infra.persistence.AuditableEntity
import jakarta.persistence.*


@Entity
class Member(
    email: String?,
    provider: OAuth2Provider,
    socialId: String,
    name: String,
    avatarUrl: String,
    id: Long = 0,
) : AuditableEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = id
        protected set

    @Column
    var email: String? = email
        protected set

    @field:Column(nullable = false, columnDefinition = "VARCHAR(20)")
    @Enumerated(EnumType.STRING)
    var provider: OAuth2Provider = provider
        protected set

    @field:Column(nullable = false)
    var socialId: String = socialId
        protected set

    @field:Column(nullable = false)
    var name: String = name
        protected set

    @field:Column(nullable = false)
    var avatarUrl: String = avatarUrl
        protected set
}